File: Assets/Scripts/Powerup.js
#pragma strict
var playerGun : GameObject;

var fireRateChange : float;
var fireRateMult : float;
var damageChange : int;
var damageMult : float;
var ammoMaxChange : int;

var gs : GunScript;
var plyr : GameObject;

function Start () 
{
	
}

function Update () 
{
	
}

function OnTriggerEnter (other : Collider) 
{
	plyr = GameObject.FindGameObjectWithTag("gun1");
	gs = plyr.gameObject.GetComponent("GunScript");
	
	gs.ammo.ammoCount += 12;
	gs.ammo.ammoCountMax += 12;

	gs.damage.damageAmount += damageChange;

	//gs.stats.isSpread = true;

	//gs.ammo.shotgunPellets = 2;

	Destroy(this.gameObject);
}

File: Assets/Scripts/Weapon/C#/PowerUp.cs
﻿using System.Collections;
using System.Collections.Generic;
using UnityEngine;

public class PowerUp : MonoBehaviour {

    GameObject playerGun;

    public float fireRateChange;
    public float fireRateMult;
    public int damageChange;
    public float damageMult;
    public int ammoMaxChange;
    public int burstChange;

    GameObject plyr;

    void OnTriggerEnter(Collider other)
    {
        plyr = GameObject.FindGameObjectWithTag("gun1");
        GunscriptCS gs = plyr.gameObject.GetComponent<GunscriptCS>();

        gs.fireRate -= fireRateChange;

        gs.damageAmount += damageChange;
        gs.burstAmnt += burstChange;

        Destroy(this.gameObject);
    }
    // Use this for initialization
    void Start () {
		
	}
	
	// Update is called once per frame
	void Update () {
		
	}
}
